requirements-optional.txt

```diff
@@ -0,0 +1 @@
+datasets>=3.6.0
```

Comment:
Rather than using a `requirements-optional.txt` we can add a section in `pyproject.toml` containing this dep, as we already do for some other optional requirements